Pod/Classes/XExtensionItem.h

```diff
@@ -96,10 +140,51 @@ typedef UIImage *(^XExtensionItemThumbnailProvidingBlock)(CGSize suggestedSize,
  */
 @property (nonatomic, copy) NSDictionary *userInfo;
 
+/**
+ Add entries from a dictionary-serializable custom object to this paramter object’s `userInfo` dictionary.
+ 
+ @param dictionarySerializable Object whose entries should be added to this paramter object’s `userInfo` dictionary.
+ */
+- (void)addEntriesToUserInfo:(id <XExtensionItemDictionarySerializing>)dictionarySerializable;
+
+#pragma mark - Initialization
+
+/**
+ Initialize an item source from a URL.
+ 
+ @param URL (Required) URL to be shared.
+ 
+ @return New item source instance.
+ */
 - (instancetype)initWithURL:(NSURL *)URL;
+
+/**
+ Initialize an item source from a string.
+ 
+ @param text (Required) Text to be shared.
+ 
+ @return New item source instance.
+ */
 - (instancetype)initWithText:(NSString *)text;
+
+/**
+ Initialize an item source from an image.
+ 
+ @param image (Required) Image to be shared.
+ 
+ @return New item source instance.
+ */
 - (instancetype)initWithImage:(UIImage *)image;
-- (instancetype)initWithData:(NSData *)data ofType:(NSString *)typeIdentifier;
+
+/**
+ Initialize an item source from data.
+ 
+ @param data           (Required) Data to be shared.
+ @param typeIdentifier (Required) Type of data.
+ 
+ @return New item source instance.
+ */
+- (instancetype)initWithData:(NSData *)data typeIdentifier:(NSString *)typeIdentifier;
```

Comment:
I've never seen typedefs used in modern Objective-C code, and it seems confusing for users of the library (it's confusing enough as is given that all of the constants are CFStringRefs rather than NSStrings).